v1.7.0/longhorn-engine/rockcraft.yaml

```diff
@@ -0,0 +1,267 @@
+
+  build-liblonghorn:
+    plugin: nil
+    after: ["setup-build-env"]
+
+    source-type: git
+    source: https://github.com/rancher/liblonghorn.git
+    source-commit: 53d1c063b95efc8d949b095bd4bf04637230265f
+    source-depth: 1
+
+    override-build: |
+      set -eux -o pipefail
+
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L15-L22
+      cd $CRAFT_PART_SRC
+      make
+      make install
+
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L24-L31
+  build-tgt:
+    plugin: nil
+    after: ["setup-build-env"]
+
+    source-type: git
+    source: https://github.com/rancher/tgt.git
+    source-commit: 3a8bc4823b5390e046f7aa8231ed262c0365c42c
+    source-depth: 1
+
+    override-build: |
+      set -eux -o pipefail
+
+      cd $CRAFT_PART_SRC
+      make
+      make install
+
+      # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L52-L56
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/bin
+      cp /usr/sbin/{tgt-admin,tgt-setup-lun,tgtadm,tgtd,tgtimg} $CRAFT_PART_INSTALL/usr/local/bin
+
+  # Pulls a pre-built GRPC health probe executable into the final ROCK.
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L33-L35
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L51
+  fetch-grpc-health-probe:
+    plugin: nil
+    after: ["setup-build-env"]
+
+    override-build: |
+      set -eux -o pipefail
+
+      VERSION="v0.4.28"
+      ARCH="$CRAFT_TARGET_ARCH"
+
+      URL="https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${VERSION}/grpc_health_probe-linux-${ARCH}"
+
+      mkdir -p $CRAFT_PART_INSTALL/usr/local/bin
+      curl -sSfL $URL -o $CRAFT_PART_INSTALL/usr/local/bin/grpc_health_probe
+      chmod +x $CRAFT_PART_INSTALL/usr/local/bin/grpc_health_probe
+
+  # Pulls a pre-built binary release of the `tini` init system.
+  # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L64-L67
+  fetch-tini-init:
+    plugin: nil
+    after: ["setup-build-env"]
+
+    override-build: |
+      set -eux -o pipefail
+
+      TINI_VERSION="v0.19.0"
+      ARCH="$CRAFT_TARGET_ARCH"
+
+      URL="https://github.com/krallin/tini/releases/download/${TINI_VERSION}/tini-${ARCH}"
+
+      curl -sSfL $URL -o $CRAFT_PART_INSTALL/tini
+      chmod +x $CRAFT_PART_INSTALL/tini
+
+  prep-final-image:
+    plugin: nil
+
+    # https://github.com/longhorn/longhorn-engine/blob/v1.7.0/package/Dockerfile#L45-L46
+    stage-packages:
+      - nfs-common
+      - nfs4-acl-tools
+      - cifs-utils
+      - libaio-dev
+      - sg3-utils
+      # NOTE(aznashwan): original image install 'iputils' SLE package
+      # which contains the utilities from all the below packages:
+      - iputils-arping
+      - iputils-ping
+      - iputils-tracepath
+      - iputils-clockdiff
+      - iproute2
+      # - qemu-tools
```

Comment:
Any reason for commenting this out?

Reply:
Great catch! I had left that as commented to check whether the whole `qemu-tools` package was actually needed or not since i had a hunch it didn't.

TL;DR: there is a `longhornctl` subcommand which calls the `qemu-img` binary, so `qemu-img` is the only thing that's needed, not that whole contents of the `qemu-tools` SLES package.

Context:
* [this Engine commit](https://github.com/longhorn/longhorn-engine/commit/54accb6a0) added the Longhorn subcommand which needs `qemu-img` and [first adds the `qemu-utils` Ubuntu 16.04](https://github.com/longhorn/longhorn-engine/commit/54accb6a0#diff-c55d1ee74f32cd4b9eeeceaec981a180f52e761acb158a5b0b0e088c7725c9caL4-R4) package dep (back then the images were based on Trusty)
* [this Engine commit](https://github.com/longhorn/longhorn-engine/commit/2b055aa80#diff-c55d1ee74f32cd4b9eeeceaec981a180f52e761acb158a5b0b0e088c7725c9caL7-R13) switched the image building from Ubuntu 16.04 to SLES BCI and thus translated the `qemu-utils` Ubuntu Deb to `qemu-tools` SLES RPM
* the [`qemu-tools` SLE package](https://opensuse.pkgs.org/15.6/opensuse-oss-x86_64/qemu-tools-8.2.2-150600.1.8.x86_64.rpm.html) contains a bunch of random binaries/scripts but depends on a separate SLE package named `qemu-img`

As such, I'll update it to simply install `qemu-utils` on Ubuntu.